deployments/extensions.go

```diff
@@ -466,6 +538,134 @@ func (e *Extension) Uninstall(c *kubernetes.Cluster, ui *ui.UI, options *kuberne
+		var buf bytes.Buffer
+		_, _ = io.Copy(&buf, resp.Body)
+
+		fmt.Printf("response body as string: \n%s\n", buf.String())
+
+		ext := registeredExtension{}
+
+		err = json.Unmarshal(buf.Bytes(), &ext)
+		if err != nil {
+			return errors.Wrap(err, "failed to parse description file")
+		}
+		ui.Exclamation().Msg(fmt.Sprintf("extension already present with id %s, product %s", *ext.ID, *ext.Product))
+	} else if resp.StatusCode == 404 {
+		ui.Exclamation().Msg(fmt.Sprintf("extension %s not yet registered, processing now", e.Name))
+	} else {
+		return errors.New(fmt.Sprintf("Unexepcted response from registry: %d", resp.StatusCode))
+	}
+	return nil
+}
+
+// Unregister extension from the extension registry
+func (e *Extension) UnRegister(c *kubernetes.Cluster, ui *ui.UI, options *kubernetes.InstallationOptions) error {
+	domain, err := options.GetString("system_domain", "")
+	if err != nil {
+		return errors.New("system_domain value not provided")
+	}
+
+	fusemlURL := fmt.Sprintf("http://%s.%s", CoreDeploymentID, domain)
+	fullURL := fmt.Sprintf("%s/extensions/%s", fusemlURL, e.Desc.Product)
+
+	// no direct Delete method so we have to create a client and a request
+	client := &http.Client{}
+
+	req, err := http.NewRequest("DELETE", fullURL, nil)
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode == 204 {
+		return nil
+	}
+
+	// no such extension found, that might be OK...
+	if resp.StatusCode == 404 {
+		return nil
+	}
+
+	fmt.Println("response from DELETE: ", resp.StatusCode)
+
+	// something else is wrong, read the response from POST call
+	var buf bytes.Buffer
+	_, _ = io.Copy(&buf, resp.Body)
+
+	return errors.New(fmt.Sprintf("Failed unregistering the extension. Server returns %s: ", buf.String()))
+}
+
+// Register extension in the registry that is run by fuseml-core server
+func (e *Extension) Register(c *kubernetes.Cluster, ui *ui.UI, options *kubernetes.InstallationOptions) error {
+
+	domain, err := options.GetString("system_domain", "")
+	if err != nil {
+		return errors.New("system_domain value not provided")
+	}
+
+	fusemlURL := fmt.Sprintf("http://%s.%s", CoreDeploymentID, domain)
+	fullURL := fmt.Sprintf("%s/extensions", fusemlURL)
+
+	extServices := []*registeredExtensionService{}
+	for _, service := range e.Desc.Services {
+		extSetvice := registeredExtensionService{
+			ID:          service.ID,
+			ExtensionID: &e.Name,
+			Resource:    service.Resource,
+			Category:    service.Category,
+			Description: service.Description,
+		}
+		extServices = append(extServices, &extSetvice)
+	}
+
+	ext := registeredExtension{
+		ID:          &e.Name,
+		Product:     &e.Desc.Product,
+		Version:     &e.Desc.Version,
+		Description: &e.Desc.Description,
+		Services:    extServices,
+	}
+
+	jsonValue, err := json.Marshal(ext)
+	if err != nil {
+		return err
+	}
+	resp, err := http.Post(fullURL, "application/json", bytes.NewBuffer(jsonValue))
```

Comment:
This is probably where you wanted to check if the extension is already registered, otherwise you'll get a Conflict response here.
You should consider the case where someone is running the installer repeatedly to install more extensions. Should an "extension already exists" response here be treated as an error ?

Reply:
OK, good point, here's probably the point where my "getExtension" function should be used.

> Should an "extension already exists" response here be treated as an error ?

Hm, what do you think? I think probably not ... for the case of running installer repeatedly it would be a bummer.

Reply:
See my 3rd commit

Reply:
Yup, I agree